File: ports/1.21.1-full/src/main/java/mods/eln/node/six/SixNodeCacheStd.java
package mods.eln.node.six;

import mods.eln.node.ISixNodeCache;
import net.minecraft.world.level.block.Block;
import net.minecraft.world.level.block.BaseEntityBlock;
import net.minecraft.world.item.ItemStack;

public class SixNodeCacheStd implements ISixNodeCache {

    @Override
    public boolean accept(ItemStack stack) {

        Block b = Block.getBlockFromItem(stack.getItem());
        if (b == null) return false;
        if (b instanceof BaseEntityBlock) return false;
        return b.getRenderType(b.getDefaultState()).equals(0) && !(stack.getItem() instanceof SixNodeItem);
    }

    @Override
    public int getMeta(ItemStack stack) {

        return stack.getItemDamage();
    }

}

File: src/main/kotlin/mods/eln/node/six/SixNodeCacheStd.kt
package mods.eln.node.six

import mods.eln.misc.Utils
import mods.eln.node.ISixNodeCache
import net.minecraft.world.level.block.Block
import net.minecraft.block.BlockContainer
import net.minecraft.world.item.ItemStack
import net.minecraft.world.level.block.RenderShape

class SixNodeCacheStd : ISixNodeCache {
    override fun accept(stack: ItemStack): Boolean {
        Utils.println("Testing item ${stack.hoverName} for blockiness")
        val b = Block.getBlockFromItem(stack.item) ?: return false
        if (b is BlockContainer) return false
        // 1.8+: the integer render types (0 cube, 31 logs, 39 quartz, 59 chisel) are all baked
        // models now; "camouflage-able" means a full-cube model.
        val state = b.defaultState
        Utils.println("Item is probably a block with render type ${state.renderType}")
        return if (stack.item is SixNodeItem) false else
            state.renderType == RenderShape.MODEL && state.isFullCube
    }

    override fun getMeta(stack: ItemStack): Int {
        return stack.itemDamage
    }
}
